### PR TITLE
implement server start-stop methods in openstack plugin

### PR DESCRIPTION
**Pull request by baixinsui**
fixes https://github.com/eclipse-xpanse/xpanse/issues/1161

### CODE REVIEW

File: plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/manage/ServersManager.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            } else {
+                log.error("Stop server with id {} failed. error:{}", resource.getResourceId(),
+                        actionResponse.getFault());
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Restart the Huawei Cloud Ecs server.
```
Comment (swaroopar) [inline plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/manage/ServersManager.java:97]: comment to be corrected.

File: plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/manage/ServersManager.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                log.error("Start server with id {} failed. error:{}", resource.getResourceId(),
+                        actionResponse.getFault());
+            }
+        }
+        return true;
+    }
+
+
+    /**
+     * Stop the Huawei Cloud Ecs server.
```
Comment (swaroopar) [inline plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/manage/ServersManager.java:74]: comment to be corrected.

File: plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/manage/ServersManager.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            ActionResponse actionResponse =
+                    osClient.compute().servers().action(resource.getResourceId(), Action.START);
+            if (actionResponse.isSuccess()) {
+                log.info("Start server with id {} successfully.", resource.getResourceId());
+            } else {
+                log.error("Start server with id {} failed. error:{}", resource.getResourceId(),
+                        actionResponse.getFault());
+            }
+        }
+        return true;
```
Comment (swaroopar) [inline plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/manage/ServersManager.java:69]: we are returning always true. Should we return false if actionReponse.isSuccess() is false?

File: runtime/src/test/java/org/eclipse/xpanse/runtime/modules/common/OpenApiGeneratorJarManageTest.java [hunk 4]
```diff
@@ -44,6 +44,7 @@ void testGetOpenApiWorkdir() {
     }
 
     @Test
+    @Disabled
```
Comment (swaroopar) [inline runtime/src/test/java/org/eclipse/xpanse/runtime/modules/common/OpenApiGeneratorJarManageTest.java:47]: should we really disable this?

**Review by swaroopar — APPROVED** [review]
LGTM

**Review by swaroopar — CHANGES_REQUESTED** [review]
comments added.

File: runtime/src/test/java/org/eclipse/xpanse/runtime/modules/common/OpenApiGeneratorJarManageTest.java (hunk 4 above)
Comment (baixinsui) [reply runtime/src/test/java/org/eclipse/xpanse/runtime/modules/common/OpenApiGeneratorJarManageTest.java:47]: Removed.

File: plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/manage/ServersManager.java (hunk 3 above)
Comment (baixinsui) [reply plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/manage/ServersManager.java:69]: Updated.

File: plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/manage/ServersManager.java (hunk 2 above)
Comment (baixinsui) [reply plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/manage/ServersManager.java:74]: Updated.

File: plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/manage/ServersManager.java (hunk 1 above)
Comment (baixinsui) [reply plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/manage/ServersManager.java:97]: Updated.

File: plugins/openstack/src/test/java/org/eclipse/xpanse/plugins/openstack/OpenstackOrchestratorPluginTest.java [hunk 5]
```diff
@@ -1,91 +1,434 @@
...
+    void testGetCsp() {
+        assertThat(openstackOrchestratorPluginUnderTest.getCsp()).isEqualTo(Csp.OPENSTACK);
     }
 
     @Test
-    void getAvailableCredentialTypes() {
-        assertEquals(List.of(CredentialType.VARIABLES), plugin.getAvailableCredentialTypes());
+    void testRequiredProperties() {
+        assertThat(openstackOrchestratorPluginUnderTest.requiredProperties())
+                .isEqualTo(List.of(OpenstackEnvironmentConstants.AUTH_URL));
```
Comment (swaroopar) [inline plugins/openstack/src/test/java/org/eclipse/xpanse/plugins/openstack/OpenstackOrchestratorPluginTest.java:68]: @baixinsui this must be an empty list. This was changed with another PR which was merged today morning.

Comment (baixinsui) [reply plugins/openstack/src/test/java/org/eclipse/xpanse/plugins/openstack/OpenstackOrchestratorPluginTest.java]: Updated.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**